From 27fae5088b20c5bb448c8449fba113f85b45c484 Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@st.com>
Date: Thu, 25 Jan 2024 14:36:31 +0100
Subject: [PATCH 470/841] feat(stm32mp2): handle st,mask-reset property

Handle st,mask-reset property.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 2e90c8ff8c11d7360e4b03de57ae6f0e372b2bd7 feat(stm32mp2): handle st,mask-reset property]
Change-Id: Ic9cf98f136f217c7dcdc4e627e03300e9a888455
Signed-off-by: Pascal Paillet <p.paillet@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/354411
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
---
 drivers/st/pmic/stm32mp_pmic2.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/st/pmic/stm32mp_pmic2.c b/drivers/st/pmic/stm32mp_pmic2.c
index a95b2c73a..fd8c5514e 100644
--- a/drivers/st/pmic/stm32mp_pmic2.c
+++ b/drivers/st/pmic/stm32mp_pmic2.c
@@ -442,6 +442,14 @@ static int register_pmic2(void)
 				regul->bypass_mv);
 		}
 
+		if (fdt_getprop(fdt, subnode, "st,mask-reset", NULL)  != NULL) {
+			VERBOSE("%s: set mask-reset\n", desc->node_name);
+			ret = pmic2_set_flag(desc, REGUL_MASK_RESET);
+			if (ret != 0) {
+				ERROR("set mask-reset failed\n");
+				return ret;
+			}
+		}
 	}
 
 	return 0;
-- 
2.47.2

